components/ILIAS/Forum/classes/Thread/ForumThreadTableSessionStorage.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+namespace ILIAS\Forum\Thread;
+
+use ForumDto;
+use ilForum;
+use ilForumAuthorInformationCache;
+use ilForumProperties;
+use ilForumTopic;
+use ILIAS\HTTP\Wrapper\WrapperFactory;
+use ILIAS\Refinery\Factory;
+use ilObjForum;
+use ilSession;
+use ThreadSortation;
+
+class ForumThreadTableSessionStorage
+{
+    public const KEY_THREAD_SORTATION = 'thread_sortation';
+    public const KEY_THREAD_PAGE = 'thread_page';
+
+    private WrapperFactory $http_wrapper;
+    private Factory $refinery;
+
+    public function __construct(
+        private int $forum_ref_id,
+        private bool $is_moderator,
+        ?WrapperFactory $http_wrapper = null,
+        ?Factory $refinery = null
+    ) {
+        $this->forum_ref_id = $forum_ref_id;
+        $this->is_moderator = $is_moderator;
+
+        global $DIC;
+        $this->http_wrapper = $http_wrapper ?? $DIC->http()->wrapper();
+        $this->refinery = $refinery ?? $DIC->refinery();
+    }
+
+    public function fetchData(ilForum $forum, ForumDto $topicData): array
```

Comment:
Can we introduce a simple object instead of the array? I suggest something like `ThreadsPage`?

We could then also get rid of the `if (array_key_exists('thread', $thread)) {` checks in class `ilObjForumGUI`.